### PR TITLE
Upgrade canvas dependency

### PR DESCRIPTION
**Pull request by marthacryan**
Fixes elyra-ai/elyra#2300
 
**Note:** There was a bug introduced in the canvas release 12.6 (open PR: https://github.com/elyra-ai/canvas/pull/1014, tracked by https://github.com/elyra-ai/canvas/issues/1011) that causes a console error to appear when dragging in JupyterLab. It doesn't break any functionality but it's preferable not to have any errors show up on the console. In this PR, I moved to version 12.5 to prevent that error but I'm planning to work on a PR in canvas to fix it. 

**Changes:**
- Adds 3 dependencies due to the changes in elyra-ai/canvas#879
- Specifies node version in code sandbox due to build error
- CI tests caught regression from #152 - not sure why this changed with the update but I added code to cover that case. 

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/package.json
Comment (ajbozarth) [inline packages/pipeline-editor/package.json:36]: We normally sort these alphabetically (not sure why the linter isn't fixing it)

File: packages/pipeline-editor/src/PipelineController/index.ts
Comment (ajbozarth) [inline packages/pipeline-editor/src/PipelineController/index.ts:601]: does this still handle the case where we set the value back to undefined (I forget which properties we do that one)?

**Review by ajbozarth — COMMENTED** [review]
A couple quick comments, I'll test it tomorrow

File: packages/pipeline-editor/src/PipelineController/index.ts
Comment (marthacryan) [reply packages/pipeline-editor/src/PipelineController/index.ts]: I refactored this a little due to a test failure, I don't think that would happen with updated code but you could take a look in case

File: packages/pipeline-editor/src/PipelineController/index.ts
Comment (ajbozarth) [inline packages/pipeline-editor/src/PipelineController/index.ts:602]: Ah I see what this is doing now. We're actually doing something similar on the Elyra side already https://github.com/elyra-ai/elyra/blob/9ea71730266eee3185631524db4f5e0873c139b3/packages/pipeline-editor/src/PipelineEditorWidget.tsx#L284-L292

But I'm unsure why this is needed with the canvas update? Is the newer version of canvas not dealing with undefined/null values?

File: packages/pipeline-editor/package.json
Comment (ajbozarth) [inline packages/pipeline-editor/package.json:24]: why lock to an older minor version? I remember you explaining it somewhere but can't find the comment

Comment (marthacryan) [reply packages/pipeline-editor/package.json:24]: There's a bug that was introduced in 12.6 that throws a console error when JL triggers a drag event. I contacted the canvas people about it but I figured for now we could just pin to the minor.

File: packages/pipeline-editor/src/PipelineController/index.ts
Comment (marthacryan) [reply packages/pipeline-editor/src/PipelineController/index.ts:602]: Not totally sure why this showed up but we had a test for it so I figured I'd just include a fix.

File: packages/pipeline-editor/package.json
Comment (ajbozarth) [reply packages/pipeline-editor/package.json:24]: I'd say we wait till they fix that to merge unless it will delay our release then

File: packages/pipeline-editor/src/PipelineController/index.ts
Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineController/index.ts:602]: super weird but whatever

File: packages/pipeline-editor/package.json
Comment (marthacryan) [reply packages/pipeline-editor/package.json:24]: I think it's worth merging without those changes because we don't really know how high of a priority this is for them and this was a high priority for the upcoming release. One note though - it doesn't actually break anything in our code, it just shows a console error

Comment (ajbozarth) [reply packages/pipeline-editor/package.json:24]: If it's just an upstream console error with no problems in elyra then I say let it in, it will go away automatically when we catch their release containing the fix

File: yarn.lock
Comment (ajbozarth) [inline yarn.lock:460]: forgot to clear and rebuild the yarn lock :)

**Review by ajbozarth — APPROVED** [review]
code-wise LGTM, is there anything I should check locally?

**Comment by marthacryan** [comment]
@ajbozarth I don't think there's much to check locally, there don't seem to be any big user-facing changes but it doesn't hurt to check

File: .codesandbox/ci.json
Comment (lresende) [inline .codesandbox/ci.json:7]: Is Node 16 a requirement? if so, should we update Elyra's readme that currently mentions Node 12+?

**Review by lresende — APPROVED** [review]

**Comment by lresende** [comment]
@marthacryan is this ready to be merged?

File: .codesandbox/ci.json
Comment (ajbozarth) [reply .codesandbox/ci.json:7]: IIRC, there was an issue in CI that caused us to need to set a version. I'm unsure if it needs to be 16 or not though.

Comment (marthacryan) [reply .codesandbox/ci.json:7]: This is just for code sandbox so it's not a new requirement for elyra.

**Comment by ajbozarth** [comment]
@marthacryan is on vacation this week, but this was just waiting for a second approval (other than mine) before we merged it. So if you're ok with the Node update then I can merge this and we can release it in an rc alongside the global properties PR

**Pull request merged**
